x-pack/solutions/observability/plugins/observability/server/services/related_dashboards_client.ts

```diff
@@ -48,18 +53,37 @@ export class RelatedDashboardsClient {
         `Alert with id ${this.alertId} not found. Could not fetch related dashboards.`
       );
     }
+    const [suggestedDashboards, linkedDashboards] = await Promise.all([
+      this.fetchSuggestedDashboards(),
+      this.getLinkedDashboards(),
+    ]);
+    const filteredSuggestedDashboards = suggestedDashboards.filter(
+      (suggested) => !linkedDashboards.some((linked) => linked.id === suggested.id)
```

Comment:
My initial impression was "why not use the set method we have below to get O(1) lookup" but I am thinking we are only going to have a handful of linked dashboards in most cases, where N=10 or less, so probably not needed. Agree? Just a thought.